pkg/services/dashboards/service/integration_test/dashboard_k8s_test.go

```diff
@@ -0,0 +1,1680 @@
+			})
+
+			// Test that editor cannot update a dashboard in the folder (should fail)
+			t.Run("Editor cannot update dashboard in restricted folder", func(t *testing.T) {
+				// Create a dashboard in the folder with admin
+				dash, err := createDashboard(t, adminClient, "Dashboard to Update in Restricted Folder", &folderUID, nil)
+				require.NoError(t, err)
+				require.NotNil(t, dash)
+
+				// Editor should not be able to update the dashboard (only has view permission)
+				_, err = updateDashboard(t, editorClient, dash, "Updated by Editor in Restricted Folder")
+				require.Error(t, err, "Should not be able to update dashboard with only VIEW permission")
+
+				// Clean up
+				err = adminClient.Resource.Delete(context.Background(), dash.GetName(), v1.DeleteOptions{})
+				require.NoError(t, err)
+			})
+
+			// Now change to EDIT permissions and verify behavior changes
+			t.Run("Change to EDIT permissions", func(t *testing.T) {
+				// Change permissions for the editor to EDIT
+				setResourceUserPermission(t, ctx, ctx.AdminUser, "folders", folderUID, editorUserID, dashboardaccess.PERMISSION_EDIT)
+
+				// Test that editor can now create a dashboard in the folder (should succeed)
+				t.Run("Editor can now create dashboard in folder", func(t *testing.T) {
+					// Create a dashboard in the folder with editor
+					dash, err := createDashboard(t, editorClient, "Editor Dashboard with EDIT Permission", &folderUID, nil)
+					require.NoError(t, err)
+					require.NotNil(t, dash)
+
+					// Verify it was created in the correct folder
+					spec := dash.Object["spec"].(map[string]interface{})
+					require.Equal(t, folderUID, spec["folderUID"])
+
+					// Clean up
+					err = adminClient.Resource.Delete(context.Background(), dash.GetName(), v1.DeleteOptions{})
+					require.NoError(t, err)
+				})
+
+				// Test that editor can now update a dashboard in the folder (should succeed)
+				t.Run("Editor can now update dashboard in folder", func(t *testing.T) {
+					// Create a dashboard in the folder with admin
+					dash, err := createDashboard(t, adminClient, "Dashboard to Update with EDIT Permission", &folderUID, nil)
+					require.NoError(t, err)
+					require.NotNil(t, dash)
+
+					// Editor should now be able to update the dashboard (has EDIT permission)
+					updatedDash, err := updateDashboard(t, editorClient, dash, "Updated by Editor with EDIT Permission")
+					require.NoError(t, err)
+					require.NotNil(t, updatedDash)
+
+					// Verify the update
+					updatedSpec := updatedDash.Object["spec"].(map[string]interface{})
+					require.Equal(t, "Updated by Editor with EDIT Permission", updatedSpec["title"])
+
+					// Clean up
+					err = adminClient.Resource.Delete(context.Background(), dash.GetName(), v1.DeleteOptions{})
+					require.NoError(t, err)
+				})
+			})
+
+			// Clean up the folder
+			// TODO: CHeck if this should be forced to admin.
```

Comment:
What user is it currently using? I'm surprised this passes if it isn't an admin or editor

Reply:
I actually did already change the folderClient to always be via the admin user, seems like I forgot to remove the TODO. Thanks!